draft-ietf-tls-esni.md

```diff
@@ -503,15 +503,31 @@ requires the "encrypted_client_hello" be computed after all other extensions.
 In particular, this is possible because the "pre_shared_key" extension is
 forbidden in ClientHelloOuter.
 
+Next, the client computes the AAD input for the HPKE sealing operation. It
+consists of the ClientHelloOuterAAD and the parameters used by the server to
+compute the HPKE context:
+
+~~~
+    ech_aad =
+        ClientECH.cipher_suite ||
+        ClientECH.config_id ||
+        ClientECH.enc ||
+        ClientHelloOuterAAD
+~~~
+
+The ClientECH fields are encoded with their length prefixes as specified by the
+ClientECH structure. In other words, the prefix of `ech_aad` is the prefix of
+the serialized ClientECH structure that includes the cipher suite, the
+configuration identifier, and the encapsulated key.
```

Comment:
I think we're better off writing this a bit more explicitly in TLS syntax, given we already have way to describe the length prefixes.

Alternatively, should we say it's the ClientHelloOuter, but with the `payload` field of the encrypted_client_hello extension replaced with the empty string?

Reply:
I think using TLS syntax is a good idea.

Reply:
Done.